benchmark/wildtime_benchmarks/example_pipelines/yearbook.yaml

```diff
@@ -41,10 +41,12 @@ data:
   dataset_id: yearbook
   transformations: []
   bytes_parser_function: |
+    import warnings
     import torch
-    import numpy as np
     def bytes_parser_function(data: memoryview) -> torch.Tensor:
-      return torch.from_numpy(np.frombuffer(data, dtype=np.float32))
+      with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=UserWarning)
+      return torch.frombuffer(data, dtype=torch.float32).reshape(3, 32, 32)
```

Comment:
I try to remember why we use torch.from_numpy and np.from_buffer instead of torch.frombuffer. From a quick test, they seem to behave identical:

```
>>> byte_array = bytearray('WXYZ', 'utf-8')
>>> mv = memoryview(byte_array)
>>> torch.frombuffer(mv, dtype=torch.float32)
tensor([1.5294e+16])
>>> torch.from_numpy(np.frombuffer(mv, dtype=np.float32))
tensor([1.5294e+16])
```

But there should be a reason I did this (we played around with it quite a bit, but tbh I do not remember right now what the problem was). If you tested this within the online dataset and it works, I guess it should be fine. I am a tad confused :D

Reply:
https://github.com/search?q=repo%3Aeth-easl%2Fmodyn%20frombuffer&type=code

For Criteo, I also used frombuffer. I am confused. What did I do here :D